src/java.base/unix/classes/sun/nio/fs/UnixSecureDirectoryStream.java

```diff
@@ -117,13 +117,18 @@ private UnixPath getName(Path obj) {
                 newdfd2 = dup(newdfd1);
                 ptr = fdopendir(newdfd1);
             } catch (UnixException x) {
-                if (newdfd1 != -1)
-                    UnixNativeDispatcher.close(newdfd1);
-                if (newdfd2 != -1)
-                    UnixNativeDispatcher.close(newdfd2);
-                if (x.errno() == UnixConstants.ENOTDIR)
-                    throw new NotDirectoryException(file.toString());
-                x.rethrowAsIOException(file);
+                IOException ioe = x.errno() == UnixConstants.ENOTDIR ?
+                    new NotDirectoryException(file.toString()) :
+                    x.asIOException(file);
+                if (newdfd1 != -1) {
+                    ioe = UnixNativeDispatcher.close(newdfd1, ioe,
+                        y -> y.asIOException(file));
+                }
+                if (newdfd2 != -1) {
+                    ioe = UnixNativeDispatcher.close(newdfd1, ioe,
+                        z -> z.asIOException(file));
+                }
```

Comment:
Same as UnixFileSystemProvider, I don't think the errors from closing newfd1 and newfd2 are useful here.